File: src/main/java/net/ndrei/teslacorelib/capabilities/wrench/ITeslaWrenchHandler.java
package net.ndrei.teslacorelib.capabilities.wrench;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.EnumActionResult;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.EnumHand;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.ndrei.teslacorelib.items.TeslaWrench;

/**
 * Created by CF on 2016-12-13.
 */
public interface ITeslaWrenchHandler {
    EnumActionResult onWrenchUse(TeslaWrench wrench,
                                 EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand,
                                 EnumFacing facing, float hitX, float hitY, float hitZ);
}

File: src/main/kotlin/net/ndrei/teslacorelib/capabilities/wrench/ITeslaWrenchHandler.kt
package net.ndrei.teslacorelib.capabilities.wrench

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.EnumActionResult
import net.minecraft.util.EnumFacing
import net.minecraft.util.EnumHand
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.ndrei.teslacorelib.items.TeslaWrench

/**
 * Created by CF on 2017-06-28.
 */
interface ITeslaWrenchHandler {
    fun onWrenchUse(wrench: TeslaWrench,
                    player: EntityPlayer, world: World, pos: BlockPos, hand: EnumHand,
                    facing: EnumFacing, hitX: Float, hitY: Float, hitZ: Float): EnumActionResult
}